File: src/main/java/me/aov/sellgui/api/SellGUIEvents.java
package me.aov.sellgui.api;

import me.aov.sellgui.SellGUI;
import org.bukkit.entity.Player;
import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;
// soon
public class SellGUIEvents extends Event {
    private static final HandlerList handlers = new HandlerList();
    private final Player player;
    private final SellGUI sellGUI;

    public SellGUIEvents(Player player, SellGUI sellGUI) {
        this.player = player;
        this.sellGUI = sellGUI;
    }

    public Player getPlayer() {
        return player;
    }

    public SellGUI getSellGUI() {
        return sellGUI;
    }

    @Override
    public HandlerList getHandlers() {
        return handlers;
    }

    public static HandlerList getHandlerList() {
        return handlers;
    }
}

File: src/main/kotlin/me/aov/sellgui/api/SellGUIEvents.kt
package me.aov.sellgui.api

import me.aov.sellgui.SellGUI
import org.bukkit.entity.Player
import org.bukkit.event.Event
import org.bukkit.event.HandlerList

open class SellGUIEvents(private val player: Player, private val sellGUI: SellGUI) : Event() {
    fun getPlayer(): Player = player

    fun getSellGUI(): SellGUI = sellGUI

    override fun getHandlers(): HandlerList = handlers

    companion object {
        private val handlers = HandlerList()

        @JvmStatic
        fun getHandlerList(): HandlerList = handlers
    }
}
